include/zephyr/net/wifi_nm_wpa_supp.h

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_ZEPHYR_NET_WIFI_NM_WPA_SUPP_H_
+#define ZEPHYR_INCLUDE_ZEPHYR_NET_WIFI_NM_WPA_SUPP_H_
+
+#include <zephyr/net/wifi_nm.h>
+
+union supp_mon_event_data {
+	struct supp_event_auth_reject {
+		int auth_type;
+		int auth_transaction;
+		int status_code;
+		uint8_t bssid[ETH_ALEN];
+	} auth_reject;
+
+	struct supp_event_connected {
+		uint8_t bssid[ETH_ALEN];
+		char ssid[SSID_MAX_LEN];
+		int id;
+	} connected;
+
+	struct supp_event_disconnected {
+		uint8_t bssid[ETH_ALEN];
+		int reason_code;
+		int locally_generated;
+	} disconnected;
+
+	struct supp_event_assoc_reject {
+		int status_code;
+		int reason_code;
+	} assoc_reject;
+
+	struct supp_event_temp_disabled {
+		int id;
+		char ssid[SSID_MAX_LEN];
+		unsigned int auth_failures;
+		unsigned int duration;
+		char reason_code[32];
+	} temp_disabled;
+
+	struct supp_event_reenabled {
+		int id;
+		char ssid[SSID_MAX_LEN];
+	} reenabled;
+
+	struct supp_event_bss_added {
+		unsigned int id;
+		uint8_t bssid[ETH_ALEN];
+	} bss_added;
+
+	struct supp_event_bss_removed {
+		unsigned int id;
+		uint8_t bssid[ETH_ALEN];
+	} bss_removed;
+
+	struct supp_event_network_added {
+		unsigned int id;
+	} network_added;
+
+	struct supp_event_network_removed {
+		unsigned int id;
+	} network_removed;
+
+	char supp_event_str[64];
+};
+
+enum event_num {
+	CONNECTED,
+	DISCONNECTED,
+	ASSOC_REJECT,
+	AUTH_REJECT,
+	TERMINATING,
+	SSID_TEMP_DISABLED,
+	SSID_REENABLED,
+	SCAN_STARTED,
+	SCAN_RESULTS,
+	SCAN_FAILED,
+	BSS_ADDED,
+	BSS_REMOVED,
+	NETWORK_NOT_FOUND,
+	NETWORK_ADDED,
+	NETWORK_REMOVED
+};
+
+struct status_data {
+	int event;
+	void *data;
```

Comment:
better to add length of the event? Though we can typcast to the struct based on `event` might be useful as an additional check?

Reply:
Done